# 사례별 시계열 데이터 계산하기



- 에볼라 최초 발병일 계산하기


1) ebola 데이터프레임의 마지막 행과 열을 5개씩만 살펴보자. ebola 데이터프레임은 데이터가 시간 역순으로 정렬되어 있다. 즉, 시간 순으로 데이터를 살펴 보려면 데이터프레임의 마지막부터 살펴봐야 한다.

In [5]:
import pandas as pd

ebola = pd.read_csv('C:/Users/이재윤/doit_pandas/data/country_timeseries.csv')
ebola['date_dt'] = pd.to_datetime(ebola['Date'])

In [6]:
print(ebola.iloc[-5:, :5])   

          Date  Day  Cases_Guinea  Cases_Liberia  Cases_SierraLeone
117  3/27/2014    5         103.0            8.0                6.0
118  3/26/2014    4          86.0            NaN                NaN
119  3/25/2014    3          86.0            NaN                NaN
120  3/24/2014    2          86.0            NaN                NaN
121  3/22/2014    0          49.0            NaN                NaN


2) 121행에서 볼 수 있듯이 에볼라가 발생하기 시작한 날은 2014년 03월 22일 입니다. 다음은 min 메서드를 사용하여 에볼라의 최초 발병일을 찾은 것이다.

In [7]:
print(ebola['date_dt'].min())
print(type(ebola['date_dt'].min()))

2014-03-22 00:00:00
<class 'pandas._libs.tslibs.timestamps.Timestamp'>


3) 에볼라 최초 발병일일을 알아 냈으니 Date 열에서 에볼라의 최초 발병일을 빼면 에볼라의 진행 정도를 알 수 있다.

In [8]:
ebola['outbreak_d'] = ebola['date_dt'] - ebola['date_dt'].min()

print(ebola[['Date', 'Day', 'outbreak_d']].head())

         Date  Day outbreak_d
0    1/5/2015  289   289 days
1    1/4/2015  288   288 days
2    1/3/2015  287   287 days
3    1/2/2015  286   286 days
4  12/31/2014  284   284 days
